### PR TITLE
Local storage adapters.

### PR DESCRIPTION
**Pull request by n1k0**
The idea is to allow using different local persistence backends in lieu of only IndexedDB (using localStorage or creating other adapters conforming to the expected interface will then become possible).

### CODE REVIEW

**Comment by n1k0** [comment]
r=? @Natim

File: .travis.yml [hunk 1]
```diff
@@ -12,3 +12,5 @@ before_install:
 - .env/bin/pip install kinto==1.3.1
 env:
 - KINTO_PSERVE_EXECUTABLE=.env/bin/pserve
+after_script:
+- npm run report-coverage
```
Comment (Natim) [inline .travis.yml]: :+1:

File: docs/extending.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  update(record) {
+    …
+  }
+
+  …
+}
+
+const kinto = new Kinto({adapter: MyAdapter});
```
Comment (Natim) [inline docs/extending.md]: We should say something like: "Then you can create the Kinto object passing you adapter and you are all set."

File: src/adapters/base.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+"use strict";
+
+export default class BaseAdapter {
+  /**
+   * Deletes every records present in the database..
+   *
+   * @return {Promise}
+   */
+  clear() {
+    throw new Error("Implement me.");
```
Comment (Natim) [inline src/adapters/base.js]: nit: would raise (Not Implemented) but this might be a Python deformation.

File: test/adapters/IDB_test.js [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+import IDB from "../../src/adapters/IDB.js";
+
+chai.use(chaiAsPromised);
+chai.should();
+chai.config.includeStack = true;
+
+const root = typeof window === "object" ? window : global;
+
+describe("adapters.IDB", () => {
```
Comment (Natim) [inline test/adapters/IDB_test.js]: Is it possible to make it so it is possible to reuse this test to make sure another adapter works?

Something like: https://github.com/mozilla-services/loop-server/blob/master/test/storage_test.js#L32
And: https://github.com/mozilla-services/loop-server/blob/master/test/storage_test.js#L1090-L1092

**Comment by Natim** [comment]
Looks good to me with nits.

File: test/adapters/IDB_test.js (hunk 4 above)
Comment (n1k0) [inline test/adapters/IDB_test.js]: I'll do that as soon as I'll add a localStorage adapter; that PR is quite large enough already :)

**Comment by n1k0** [comment]
I'm gonna land this and start impelmenting a localStorage adapter.

**Pull request merged**